### DISCUSSION TITLE
Clone Client to AsyncClient

### DISCUSSION THREAD

**Discussion by MajorDallas** [encode/httpx#3267] [Q&A]
My use-case is that I'm stuck using a particular library that's built on `httpx.Client`, but there are cases where I need to make dozens or hundreds of calls with this library that would greatly benefit from concurrency. I _could_ (probably) use `gevent` to get concurrency without doing anything to my code, but it would be cool if I could also do something like

```python
async with httpx.AsyncClient.from_client(lib_object._client) as client:
    ...
```

I looked through the docs and searched here hoping something like that already exists. Did I miss it?

**Comment by rafalkrupinski** [comment]
You could probably inherit from Client and implement its methods using `AsyncClient` and `asyncio.get_running_loop()` or its `anyio` equivalent
